specification/mixedreality/data-plane/Microsoft.MixedReality/preview/2021-01-01-preview/mr-arr.json

```diff
@@ -901,17 +901,7 @@
           "$ref": "#/definitions/conversion_settings"
         },
         "output": {
-          "type": "object",
-          "description": "Information about the output of a successful conversion. Only present when the status of the conversion is 'Succeeded'.",
-          "properties": {
-            "outputAssetUri": {
-              "type": "string",
-              "description": "URI of the asset generated by the conversion process.",
-              "format": "uri",
-              "example": "https://contoso.blob.core.windows.net/arr/architecture-output/house.arrAsset",
-              "readOnly": true
-            }
-          },
+          "$ref": "#/definitions/conversion_output",
           "readOnly": true
```

Comment:
I think part of the problem is this attribute, it can't be a sibling of a `$ref`.

Reply:
so should i move the readonly flag onto the $ref? can i even do this? could i drop the readOnly flag - we need to get this sorted really soon since we need to take this spec and create a final version from it. any advise is welcome. 

If this does not go through i will not do the change and have lots of pain in my SDK to rename the generated conversion output class

Reply:
It's currently on the $ref and the type, you should leave it on the type and drop it from the $ref.

Reply:
ah - okay so i completely missed that. will update!

Reply:
@jhendrixMSFT i am still confused. so readOnly cannot be a sibling of a ref - okay. But i don't see the readOnly on the type? there is a readOnly property on the "outputAssetUri" property of the conversion_output definition (but that is nested). 
As i said most likely i am missing something. Would it be enough to remove line 915 containing the "readOnly": true?

Reply:
Oh yes you're correct, I mistakenly took the `readOnly` on line 941 being applied on `conversion_output`.  I don't think you can define `conversion_output` itself as `readOnly` as that's only applicable to the `properties` section.

Let me take a step back.  Why is this being split out?  It's only referenced in one place.

Reply:
because i want to rename the generated class in the python SDK via an autorest derective. 
and only the class and not the property. 
```` yaml
directive:
    from: swagger-document
    where: '$.definitions.conversion_output'
    transform: >
        $["x-ms-client-name"] = "asset_conversion_output";
````

Reply:
I see.

Have you looked at using [`x-ms-client-name`](https://github.com/Azure/autorest/tree/master/docs/extensions#x-ms-client-name) to rename this instead?  I believe it would allow you to keep the definition inline so the `readOnly` modifier isn't lost.